tests/L0/run_transformer/run_bert_minimal_test.py

```diff
@@ -0,0 +1,168 @@
+from apex.transformer.testing.commons import print_separator
+
+mode = None
+MANUAL_SEED = 42
+inds = None
+masks = None
+data_idx = 0
+MASK_PROB = 0.1
+EASY_MODE = False
+EASY_MODE_SIZ = 32
+ONCE = False
+
+# download a public domain book as corpus
+def download_fancy_data():
+  import requests
+  response = requests.get('https://www.gutenberg.org/files/1342/1342-0.txt')
+  #response = requests.get('https://www.gutenberg.org/files/84/84-0.txt')
+  text = ' '.join(response.text.split())
+  encoded = text.encode('ascii', 'replace')
+  ints = [int(encoded[i]) for i in range(len(encoded))]
+  return torch.tensor(ints)
+
+
+# build a batch given sequence_len and batch size
+def generate_fancy_data_labels(sequence_len, batch_size):
+  global data_idx
+  global inds
+  global masks
+  global MANUAL_SEED
+  temps = list()
+  for i in range(batch_size):
+     if inds is None or data_idx >= len(inds):
+       # hack as use of RNG will fall out of sync due to pipelines being different
+       torch.manual_seed(MANUAL_SEED)
+       inds = torch.randperm(effective_length, device='cuda')
+       masks = (torch.rand(len(inds)//batch_size + 1, batch_size, sequence_len, device='cuda') >= MASK_PROB).long()
+       MANUAL_SEED += 1
+       print("new epoch", len(inds))
+       data_idx = 0
+       print("my start", inds[0:5])
+       print("masks_checksum:", torch.sum(masks))
+     if EASY_MODE:
+       data_idx_ = data_idx % EASY_MODE_SIZ
+     else:
+       data_idx_ = data_idx
+     offset = inds[data_idx_] #* SEQUENCE_LEN
+     data_idx += 1
+      
+     curr = fancy_data[offset:offset+sequence_len].clone().detach()
+     temps.append(curr)
+  temp = torch.stack(temps, dim=0).cuda()
+  mask = masks[data_idx//batch_size]
+  mask_not = torch.logical_not(mask)
+  data = mask * temp + mask_not*124
+  label = temp
+  return (data, label, mask_not)
+
+easy_data = None
+
+def fwd_step_func(batch, model):
+    global ONCE
+    data, label, loss_mask = batch
+    data = data.cuda()
+    label = label.cuda()
+    loss_mask = loss_mask.cuda()
+    y = model(data, torch.ones_like(data), lm_labels=label)
+
+    def loss_func(output_tensor):
+        output_tensor, _ = output_tensor
+        lm_loss_ = output_tensor.float()
+        lm_loss = torch.sum(
+                        lm_loss_.view(-1) * loss_mask.reshape(-1)) / loss_mask.sum()
+        averaged_loss = average_losses_across_data_parallel_group([lm_loss])
+        if data_idx >= 1536:
+            assert lm_loss < 4.8
+            if not ONCE:
+                print("LOSS OK")
+                ONCE = True
+        return lm_loss, {'avg': averaged_loss}
+    return y, loss_func
+
+
+def train(model, optim):
+    sequence_len = global_vars.get_args().seq_length
+    micro_batch_size = global_vars.get_args().micro_batch_size
+    hidden_size = global_vars.get_args().hidden_size
+    forward_backward_func = _forward_backward_pipelining_with_interleaving
```

Comment:
not a big deal, but how about calling `apex.transformer.pipeline_parallel.get_forward_backward_func` to get this interleaving schedule function?

- it's exposed here https://github.com/NVIDIA/apex/blob/25bfcb914796c7955956fc1e39839e7efed997d5/apex/transformer/pipeline_parallel/__init__.py#L6
- signature is https://github.com/NVIDIA/apex/blob/25bfcb914796c7955956fc1e39839e7efed997d5/apex/transformer/pipeline_parallel/schedules/__init__.py#L16-L18